### PR TITLE
feat: add VoyageAI embeddings

### PR DESCRIPTION
**Pull request by fzowl**
Adding VoyageAI embeddings 
Voyage AI’s embedding models and rerankers are state-of-the-art in retrieval accuracy.

### CODE REVIEW

File: docs/source/core/embedding.rst
Comment (MthwRobinson) [inline docs/source/core/embedding.rst:231]: Our documentation has moved to https://docs.unstructured.io. We can do a separate PR into https://github.com/Unstructured-IO/docs/ with these updates. cc @cmscmadd @MKhalusova

File: requirements/ingest/embed-voyageai.txt
Comment (MthwRobinson) [inline requirements/ingest/embed-voyageai.txt:2]: Our dependencies are compiled using Python 3.9 to make sure we're compatible with 3.9 - 3.12. Could you rerun `make pip-compile` from the root directory of the project using Python 3.9?

**Review by MthwRobinson — COMMENTED** [review]
Thanks @fzowl! PR looks great, just a couple of minor comments.

**Comment by fzowl** [comment]
@MthwRobinson Added a short summary, can you please take a look?

**Comment by MthwRobinson** [comment]
Summary looks good. Could you remove the `.rst` file you created for documentation and add docs in https://github.com/Unstructured-IO/docs? The `docs` directory in this repo is deprecated now in favor of the new one.

**Comment by fzowl** [comment]
@MthwRobinson I removed the .rst file and will open a PR in the docs repo soon.

**Comment by MKhalusova** [comment]
We can merge the docs after this PR is merged.

**Review by MthwRobinson — APPROVED** [review]
LGTM if tests pass!

**Comment by MthwRobinson** [comment]
@fzowl - Looks like there are some merge conflicts to clean up, but otherwise good to go

**Comment by fzowl** [comment]
@MthwRobinson I resolved the conflicts, but please check the CHANGELOG.md file

**Comment by MthwRobinson** [comment]
CHANGELOG entry looks good. For those test failures, `make tidy` and `make pip-compile` should take care of those.

**Comment by fzowl** [comment]
@MthwRobinson Ok, i run `make tidy` and make `pip-compile` :-)

**Comment by fzowl** [comment]
@MthwRobinson corrected the same

**Comment by fzowl** [comment]
@MthwRobinson What is the next step here? I mean, i see that the auto-merge is enabled now, but the merging is blocked still and `3 workflows awaiting approval`. Is there anything else i should do, is there anything open still?

**Comment by MthwRobinson** [comment]
@fzowl - Merged into a feature branch and will try to fix those dependency issues in #3099

**Comment by fzowl** [comment]
@MthwRobinson Thank you very much for all your help!

**Pull request merged**
